### PR TITLE
sample TSX UI

### PR DESCRIPTION
**Pull request by mmoskal**
It would need a lot more work. Also unclear if building tree with lambdas is a good idea.

### CODE REVIEW

**Comment by pelikhan** [comment]
This is fascinating!

File: packages/ui/src/ui.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+export class Fragment extends ParentElement<JSX.BaseProps> {
+    render(ctx: ImageContext): void {
+        this.renderChildren(ctx)
+    }
+}
+
+
+export function HorizontalGauge(props: { w: number, h: number, used: number }): JSX.Element {
+    const { w, h, used } = props
```
Comment (pelikhan) [inline packages/ui/src/ui.tsx:67]: react typically does not use acornyms, so w -> width, h -> height will feel more natural

File: packages/ui/src/ui.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    render(ctx: ImageContext): void {
+        ctx.save()
+        ctx.translate(this.x, this.y)
+        this.renderChildren(ctx)
+        ctx.restore()
+    }
+}
+
+export class Text extends BaseElement<{ children: string }> {
+    children: string
```
Comment (pelikhan) [inline packages/ui/src/ui.tsx:37]: what about optional x,y in the base element?

Comment (mmoskal) [reply packages/ui/src/ui.tsx:46]: My idea was to use `<Translate>` on elements to move them; this allows for `<Columns>`, `<Rows>` or similar in future that lays out elements (that would require some size computation going up the tree before-hands). But I guess for now having to use `<Translate>` everywhere is annoying?

Comment (pelikhan) [reply packages/ui/src/ui.tsx:46]: It can be annoying...

**Comment by pelikhan** [comment]
@mmoskal i've unbumbled it and made it a sample.

**Comment by pelikhan** [comment]
Sample tested on picobricks

**Pull request merged**